### DISCUSSION TITLE
Automatically generate new image-formats for already uploaded media

### DISCUSSION THREAD

**Discussion by 0xjasonlang** [sulu/sulu#6435] [Ideas]
If you add a new format to image-formats.xml, none of the already uploaded images get the format added so you have to manually crop or reupload them.

**Comment by alexander-schranz** [comment]
@0xjasonlang Thx for using Github Discussions.

Sulu never generates image formats on upload. It will generate them on request, so if you add a new image format you just need to clear the `Symfony Container Cache` with `bin/websiteconsole cache:clear` and the new format should be available in your twig files. For the admin the admin cache with `bin/adminconsole cache:clear`. You can also only recommended on your development machine just delete the `var/cache` directory.

Also make sure that you did configure the VHost the way Sulu has Document it:

 - https://docs.sulu.io/en/latest/cookbook/web-server/apache.html
 - https://docs.sulu.io/en/latest/cookbook/web-server/nginx.html
 
When using the Symfony CLI for your Development there is no special configuration needed as the Symfony CLI already handles it correctly. When using the `PHP Internal Webserver` with: `php -S` you need to use our provided router.php (see also [Getting Started Docs](https://docs.sulu.io/en/2.4/book/getting-started.html)):

```bash
php -S localhost:8000 -t public/ config/router.php
```

In a more detailed way, the image processing work the following way, and if your Webserver is not configured correctly the "if not exist" -> internal "redirect to controller" is not done and so you will get 404 instead of that sulu will generate the image format:

![image-formats](https://user-images.githubusercontent.com/1698337/147095663-b8b3f2c2-203d-4792-9318-0ce4df2aea89.png)

For cases were you are editing exist formats you can also use to regenerate images when changing as example quality or other metrics:

```php
bin/console sulu:media:regenerate-formats
```

This will only generate exist generated images and not generate all images, as from our point of view generate an image in all formats would be an overhead and waste of storage.

Please check your VHost configuration or let us know how you did configure it. And maybe try out the PHP Internal Webserver if it works there.
